src/BenchmarkDotNet/Order/DefaultOrderer.cs

```diff
@@ -153,6 +165,7 @@ private class BenchmarkComparer : IComparer<BenchmarkCase>
             private static readonly BenchmarkLogicalGroupRule[] DefaultOrder =
             {
                 BenchmarkLogicalGroupRule.ByCategory,
+                BenchmarkLogicalGroupRule.ByType,
```

Comment:
ByType should be first by default, I think.

Reply:
I don't know :)

I reordered it and the output didn't change for the tests.

I tried to sort it twice - all tests are passed:
```C#
[GroupBenchmarksBy(BenchmarkLogicalGroupRule.ByType)]
[GroupBenchmarksBy(BenchmarkLogicalGroupRule.ByMethod)]
class A { }
```

I did it for
```
1) for each rule + ByType
2) for each rule + ByCategory
3) for each rule + ByMethod
```
but the output is same

Reply:
Oh, stop

Reply:
Everything is right, nothing has changed 🤔

Reply:
@YegorStepanov What type are you testing? Does the type have enough options that the table should actually be different for each group rule-set?

Reply:
@timcassell 

<details>

Add these classes to the MarkdownExporterMultipleClassApprovalTests

Each rule is added in the test
```
config = config.AddLogicalGroupRules(logicalRule);
```

```C#
            public static class TwoBaselinesWithJobs111
            {
                [GroupBenchmarksBy(BenchmarkLogicalGroupRule.ByCategory)]
                [LogicalGroupColumn, CategoriesColumn, BaselineColumn]
                [BenchmarkCategory("A")]
                [SimpleJob(id: "Job1"), SimpleJob(id: "Job2")]
                public class Bench1
                {
                    [Params(10, 20)] public int Param;
                    [Benchmark(Baseline = true)] public void Foo() { }
                    [Benchmark] public void Bar() { }
                }

                [GroupBenchmarksBy(BenchmarkLogicalGroupRule.ByCategory)]
                [BenchmarkCategory("B")]
                public class Bench2
                {
                    [Params(10, 20)] public int Param;
                    [Benchmark(Baseline = true)] public void Foo() { }
                    [Benchmark] public void Bar() { }
                }
            }

            public static class TwoBaselinesWithJobs222
            {
                [GroupBenchmarksBy(BenchmarkLogicalGroupRule.ByType)]
                [LogicalGroupColumn, CategoriesColumn, BaselineColumn]
                [BenchmarkCategory("A")]
                [SimpleJob(id: "Job1"), SimpleJob(id: "Job2")]
                public class Bench1
                {
                    [Params(10, 20)] public int Param;
                    [Benchmark(Baseline = true)] public void Foo() { }
                    [Benchmark] public void Bar() { }
                }

                [GroupBenchmarksBy(BenchmarkLogicalGroupRule.ByType)]
                [BenchmarkCategory("B")]
                public class Bench2
                {
                    [Params(10, 20)] public int Param;
                    [Benchmark(Baseline = true)] public void Foo() { }
                    [Benchmark] public void Bar() { }
                }
            }

            public static class TwoBaselinesWithJobs333
            {
                [GroupBenchmarksBy(BenchmarkLogicalGroupRule.ByMethod)]
                [LogicalGroupColumn, CategoriesColumn, BaselineColumn]
                [BenchmarkCategory("A")]
                [SimpleJob(id: "Job1"), SimpleJob(id: "Job2")]
                public class Bench1
                {
                    [Params(10, 20)] public int Param;
                    [Benchmark(Baseline = true)] public void Foo() { }
                    [Benchmark] public void Bar() { }
                }

                [GroupBenchmarksBy(BenchmarkLogicalGroupRule.ByMethod)]
                [BenchmarkCategory("B")]
                public class Bench2
                {
                    [Params(10, 20)] public int Param;
                    [Benchmark(Baseline = true)] public void Foo() { }
                    [Benchmark] public void Bar() { }
                }
            }
```

</details>

Reply:
I would expect `TwoBaselinesWithJobs111` and `TwoBaselinesWithJobs222` to be the same. And `TwoBaselinesWithJobs333` should order by Type first. For reasons I explained in my other comment.

Reply:
Tbh, I'm starting to think `ByType` shouldn't even be override-able. Each Type is meant to be benchmarking a different function, so why would they be grouped any other way?

Reply:
The output for `TwoBaselinesWithJobs111` and `TwoBaselinesWithJobs222` is the same (they are slightly differs in the `LogicalGroup` column).

<details>

# ByCategory:
## TwoBaselinesWithJobs111
```
   Type | Method |        Job | Categories | Param |     Mean |   Error |  StdDev | Ratio | RatioSD |                     LogicalGroup | Baseline |
------- |------- |----------- |----------- |------ |---------:|--------:|--------:|------:|--------:|--------------------------------- |--------- |
 Bench1 |    Foo |       Job1 |          A |    10 | 102.0 ns | 6.09 ns | 1.58 ns |  1.00 |    0.00 |       A-A-Bench1-[Param=10]-Job1 |      Yes | ^
 Bench1 |    Bar |       Job1 |          A |    10 | 202.0 ns | 6.09 ns | 1.58 ns |  1.98 |    0.02 |       A-A-Bench1-[Param=10]-Job1 |       No |
        |        |            |            |       |          |         |         |       |         |                                  |          |
 Bench1 |    Foo |       Job2 |          A |    10 | 302.0 ns | 6.09 ns | 1.58 ns |  1.00 |    0.00 |       A-A-Bench1-[Param=10]-Job2 |      Yes |
 Bench1 |    Bar |       Job2 |          A |    10 | 402.0 ns | 6.09 ns | 1.58 ns |  1.33 |    0.00 |       A-A-Bench1-[Param=10]-Job2 |       No |
        |        |            |            |       |          |         |         |       |         |                                  |          |
 Bench1 |    Foo |       Job1 |          A |    20 | 502.0 ns | 6.09 ns | 1.58 ns |  1.00 |    0.00 |       A-A-Bench1-[Param=20]-Job1 |      Yes | ^
 Bench1 |    Bar |       Job1 |          A |    20 | 602.0 ns | 6.09 ns | 1.58 ns |  1.20 |    0.00 |       A-A-Bench1-[Param=20]-Job1 |       No |
        |        |            |            |       |          |         |         |       |         |                                  |          |
 Bench1 |    Foo |       Job2 |          A |    20 | 702.0 ns | 6.09 ns | 1.58 ns |  1.00 |    0.00 |       A-A-Bench1-[Param=20]-Job2 |      Yes |
 Bench1 |    Bar |       Job2 |          A |    20 | 802.0 ns | 6.09 ns | 1.58 ns |  1.14 |    0.00 |       A-A-Bench1-[Param=20]-Job2 |       No |
        |        |            |            |       |          |         |         |       |         |                                  |          |
 Bench2 |    Foo | DefaultJob |          B |    10 | 102.0 ns | 6.09 ns | 1.58 ns |  1.00 |    0.00 | B-B-Bench2-[Param=10]-DefaultJob |      Yes | ^
 Bench2 |    Bar | DefaultJob |          B |    10 | 202.0 ns | 6.09 ns | 1.58 ns |  1.98 |    0.02 | B-B-Bench2-[Param=10]-DefaultJob |       No |
        |        |            |            |       |          |         |         |       |         |                                  |          |
 Bench2 |    Foo | DefaultJob |          B |    20 | 302.0 ns | 6.09 ns | 1.58 ns |  1.00 |    0.00 | B-B-Bench2-[Param=20]-DefaultJob |      Yes | ^
 Bench2 |    Bar | DefaultJob |          B |    20 | 402.0 ns | 6.09 ns | 1.58 ns |  1.33 |    0.00 | B-B-Bench2-[Param=20]-DefaultJob |       No |
```
## TwoBaselinesWithJobs222
```
   Type | Method |        Job | Categories | Param |     Mean |   Error |  StdDev | Ratio | RatioSD |                   LogicalGroup | Baseline |
------- |------- |----------- |----------- |------ |---------:|--------:|--------:|------:|--------:|------------------------------- |--------- |
 Bench1 |    Foo |       Job1 |          A |    10 | 102.0 ns | 6.09 ns | 1.58 ns |  1.00 |    0.00 |       A-Bench1-[Param=10]-Job1 |      Yes | ^
 Bench1 |    Bar |       Job1 |          A |    10 | 202.0 ns | 6.09 ns | 1.58 ns |  1.98 |    0.02 |       A-Bench1-[Param=10]-Job1 |       No |
        |        |            |            |       |          |         |         |       |         |                                |          |
 Bench1 |    Foo |       Job2 |          A |    10 | 302.0 ns | 6.09 ns | 1.58 ns |  1.00 |    0.00 |       A-Bench1-[Param=10]-Job2 |      Yes |
 Bench1 |    Bar |       Job2 |          A |    10 | 402.0 ns | 6.09 ns | 1.58 ns |  1.33 |    0.00 |       A-Bench1-[Param=10]-Job2 |       No |
        |        |            |            |       |          |         |         |       |         |                                |          |
 Bench1 |    Foo |       Job1 |          A |    20 | 502.0 ns | 6.09 ns | 1.58 ns |  1.00 |    0.00 |       A-Bench1-[Param=20]-Job1 |      Yes | ^
 Bench1 |    Bar |       Job1 |          A |    20 | 602.0 ns | 6.09 ns | 1.58 ns |  1.20 |    0.00 |       A-Bench1-[Param=20]-Job1 |       No |
        |        |            |            |       |          |         |         |       |         |                                |          |
 Bench1 |    Foo |       Job2 |          A |    20 | 702.0 ns | 6.09 ns | 1.58 ns |  1.00 |    0.00 |       A-Bench1-[Param=20]-Job2 |      Yes |
 Bench1 |    Bar |       Job2 |          A |    20 | 802.0 ns | 6.09 ns | 1.58 ns |  1.14 |    0.00 |       A-Bench1-[Param=20]-Job2 |       No |
        |        |            |            |       |          |         |         |       |         |                                |          |
 Bench2 |    Foo | DefaultJob |          B |    10 | 102.0 ns | 6.09 ns | 1.58 ns |  1.00 |    0.00 | B-Bench2-[Param=10]-DefaultJob |      Yes | ^
 Bench2 |    Bar | DefaultJob |          B |    10 | 202.0 ns | 6.09 ns | 1.58 ns |  1.98 |    0.02 | B-Bench2-[Param=10]-DefaultJob |       No |
        |        |            |            |       |          |         |         |       |         |                                |          |
 Bench2 |    Foo | DefaultJob |          B |    20 | 302.0 ns | 6.09 ns | 1.58 ns |  1.00 |    0.00 | B-Bench2-[Param=20]-DefaultJob |      Yes | ^
 Bench2 |    Bar | DefaultJob |          B |    20 | 402.0 ns | 6.09 ns | 1.58 ns |  1.33 |    0.00 | B-Bench2-[Param=20]-DefaultJob |       No |

```
## TwoBaselinesWithJobs333
```
   Type | Method |        Job | Categories | Param |     Mean |   Error |  StdDev | Ratio | RatioSD |                       LogicalGroup | Baseline |
------- |------- |----------- |----------- |------ |---------:|--------:|--------:|------:|--------:|----------------------------------- |--------- |
 Bench1 |    Foo |       Job1 |          A |    10 | 102.0 ns | 6.09 ns | 1.58 ns |  1.00 |    0.00 |       A-Foo-Bench1-[Param=10]-Job1 |      Yes | ^
        |        |            |            |       |          |         |         |       |         |                                    |          |
 Bench1 |    Foo |       Job2 |          A |    10 | 302.0 ns | 6.09 ns | 1.58 ns |  1.00 |    0.00 |       A-Foo-Bench1-[Param=10]-Job2 |      Yes |
        |        |            |            |       |          |         |         |       |         |                                    |          |
 Bench1 |    Foo |       Job1 |          A |    20 | 502.0 ns | 6.09 ns | 1.58 ns |  1.00 |    0.00 |       A-Foo-Bench1-[Param=20]-Job1 |      Yes | ^
        |        |            |            |       |          |         |         |       |         |                                    |          |
 Bench1 |    Foo |       Job2 |          A |    20 | 702.0 ns | 6.09 ns | 1.58 ns |  1.00 |    0.00 |       A-Foo-Bench1-[Param=20]-Job2 |      Yes |
        |        |            |            |       |          |         |         |       |         |                                    |          |
 Bench1 |    Bar |       Job1 |          A |    10 | 202.0 ns | 6.09 ns | 1.58 ns |     ? |       ? |       A-Bar-Bench1-[Param=10]-Job1 |       No | ^
        |        |            |            |       |          |         |         |       |         |                                    |          |
 Bench1 |    Bar |       Job2 |          A |    10 | 402.0 ns | 6.09 ns | 1.58 ns |     ? |       ? |       A-Bar-Bench1-[Param=10]-Job2 |       No |
        |        |            |            |       |          |         |         |       |         |                                    |          |
 Bench1 |    Bar |       Job1 |          A |    20 | 602.0 ns | 6.09 ns | 1.58 ns |     ? |       ? |       A-Bar-Bench1-[Param=20]-Job1 |       No | ^
        |        |            |            |       |          |         |         |       |         |                                    |          |
 Bench1 |    Bar |       Job2 |          A |    20 | 802.0 ns | 6.09 ns | 1.58 ns |     ? |       ? |       A-Bar-Bench1-[Param=20]-Job2 |       No |
        |        |            |            |       |          |         |         |       |         |                                    |          |
 Bench2 |    Foo | DefaultJob |          B |    10 | 102.0 ns | 6.09 ns | 1.58 ns |  1.00 |    0.00 | B-Foo-Bench2-[Param=10]-DefaultJob |      Yes | ^
        |        |            |            |       |          |         |         |       |         |                                    |          |
 Bench2 |    Foo | DefaultJob |          B |    20 | 302.0 ns | 6.09 ns | 1.58 ns |  1.00 |    0.00 | B-Foo-Bench2-[Param=20]-DefaultJob |      Yes | ^
        |        |            |            |       |          |         |         |       |         |                                    |          |
 Bench2 |    Bar | DefaultJob |          B |    10 | 202.0 ns | 6.09 ns | 1.58 ns |     ? |       ? | B-Bar-Bench2-[Param=10]-DefaultJob |       No | ^
        |        |            |            |       |          |         |         |       |         |                                    |          |
 Bench2 |    Bar | DefaultJob |          B |    20 | 402.0 ns | 6.09 ns | 1.58 ns |     ? |       ? | B-Bar-Bench2-[Param=20]-DefaultJob |       No | ^
```

</details>

Reply:
Do we need a `ByRuntime` grouping? 

`ByJob` has `EnvironmentCharacteristic`, which has `RuntimeCharacteristic`. 
It seems that `ByJob` should do your desired grouping.